src/ansi-c/cprover_builtin_headers.h

```diff
@@ -2,6 +2,7 @@ void __CPROVER_assume(__CPROVER_bool assumption);
 void __VERIFIER_assume(__CPROVER_bool assumption);
 void __CPROVER_assert(__CPROVER_bool assertion, const char *description);
 void __CPROVER_precondition(__CPROVER_bool precondition, const char *description);
+void __CPROVER_postcondition(__CPROVER_bool assertion, const char *description);
```

Comment:
Random place for a comment: I think this strictly requires documentation. When should users make use of this? I note that the commit message doesn't have a body either, so it's really hard to see when that should be used.

Reply:
Thanks. Where is a good place to add documentation for this?

Reply:
As discussed elsewhere: `doc/cprover-manual/cbmc-assertions.md` is likely the best place for this. @kroening Would you be able to contribute some documentation of `__CPROVER_precondition`, which AFAIK never got documented?

Reply:
I would structure this differently.
`__CPROVER_precondition` was introduced as a means to add preconditions to library functions, i.e., it was never meant to be used by anyone but CBMC developers.

The new feature, which is contract checking, is a different use-case, and externalises this to users of CBMC. My experience is that users like manuals that are structured by use-case. Thus, I would suggest to add a new chapter, say titled "Modular Verification using Code Contracts", or the like, to discuss this use-case separately.

Reply:
> [...] never meant to be used by anyone but CBMC developers.

While this is a good intention, I don't think it will remain as constrained if people find it useful in other places as well. Anything that _can_ be used _will_ be used at some point...